kernel/src/fs/manager.rs

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use alloc::sync::{Arc, Weak};
+
+use aster_systree::{
+    impl_cast_methods_for_branch, Error, Result, SysAttrSet, SysBranchNode, SysBranchNodeFields,
+    SysNode, SysNodeId, SysNodeType, SysObj, SysPerms, SysStr,
+};
+use inherit_methods_macro::inherit_methods;
+use ostd::mm::{VmReader, VmWriter};
+use spin::Once;
+
+/// Returns a reference to the [`FsManager`]. Panics if not initialized.
+pub fn singleton() -> &'static Arc<FsManager> {
+    MANAGER.get().expect("Fs manager has not been initialized")
+}
+
+/// Initializes the [`FsManager`] singleton
+pub fn init() {
+    MANAGER.call_once(FsManager::new);
+    aster_systree::singleton()
+        .root()
+        .add_child(singleton().clone())
+        .expect("Failed to add fs manager to SysTree");
+}
+
+static MANAGER: Once<Arc<FsManager>> = Once::new();
+
+/// Manager for virtual filesystem.
+///
+/// Represents the root of all filesystem-related sysfs entries, where:
+/// - Each child node corresponds to a filesystem type (ext2, cgroup, etc.).
+/// - Manages registration and lifetime of filesystem control interfaces.
+#[derive(Debug)]
+pub struct FsManager {
+    fields: SysBranchNodeFields<dyn FsFactory>,
+    weak_self: Weak<Self>,
+}
+
+/// Trait representing a filesystem's control interface.
+pub trait FsFactory: SysObj {
+    /// Called when the interface is mounted in sysfs.
+    fn on_mount(&self);
```

Comment:
I guess that these `on_mount` and `on_unmount` methods are meant to be some kind of placeholders for a future correct implementation. If so, simply remove them. The signature of these methods can't be correct. Keeping them here would be misleading.

We obviously need some kind of FS registration mechanism (as a [previous failed attempt](https://github.com/asterinas/asterinas/pull/2130) suggested). This mechanism is essential to support Both `/proc/filesystems`, `/sys/fs/`, and the `mount` system call. But this is beyond the scope of this PR.